gen-json-schema-class.py

```diff
@@ -0,0 +1,225 @@
+OUT_DIR     = Path("registered-json-schemas")
+
+BUILD_DIR.mkdir(parents=True, exist_ok=True)
+OUT_DIR.mkdir(parents=True, exist_ok=True)
+
+# ─── LOAD MASTER SCHEMA ───────────────────────────────────────────────────────
+master      = yaml.safe_load(SCHEMA_YAML.read_text(encoding="utf-8"))
+all_classes = master.get("classes", {})
+all_slots   = master.get("slots", {})
+all_enums   = master.get("enums", {})
+
+# ─── UTILITIES ────────────────────────────────────────────────────────────────
+def collect_hierarchy(cls_name):
+    hierarchy = []
+    cur = cls_name
+    while cur:
+        cdef = all_classes.get(cur)
+        if not cdef:
+            print(f"⚠️ Warning: class '{cur}' not found; skipping")
+            break
+        hierarchy.append((cur, cdef))
+        cur = cdef.get("is_a")
+    return hierarchy
+
+
+def slot_enums(sdef):
+    enums = set()
+    if "range" in sdef:
+        enums.add(sdef["range"])
+    if "any_of" in sdef:
+        enums.update(alt["range"] for alt in sdef["any_of"] if "range" in alt)
+    return enums
+
+# ─── STEP 1: GENERATE CLASS-SPECIFIC YAML ─────────────────────────────────────
+for cls_name in all_classes:
+    print(f"\n🔨 Generating YAML for class: {cls_name}")
+    hierarchy   = collect_hierarchy(cls_name)
+    used_slots  = {}
+    used_enums  = set()
+    for _, cdef in hierarchy:
+        for slot in (cdef.get("slots") or []):
+            sdef = all_slots.get(slot)
+            if sdef:
+                used_slots[slot] = sdef
+                used_enums |= slot_enums(sdef)
+    collected_enums = {nm: all_enums[nm] for nm in used_enums if nm in all_enums}
+
+    header = yaml.safe_load(HEADER_YAML.read_text())
+    header.pop("classes", None)
+
+    yaml_content = {
+        **header,
+        "name": cls_name,
+        "classes": {
+            nm: {
+                "description": cdef.get("description"),
+                **({"is_a": cdef["is_a"]} if "is_a" in cdef else {}),
+                "slots": cdef.get("slots") or []
+            }
+            for nm, cdef in hierarchy
+        },
+        "slots": used_slots,
+        "enums": collected_enums
+    }
+
+    out_file = BUILD_DIR / f"{cls_name}.yaml"
+    out_file.write_text(yaml.safe_dump(yaml_content, sort_keys=False))
+    print(f"  • Wrote {out_file}")
+
+# ─── STEP 2 + 3: JSON SCHEMA GENERATION + DEREFERENCE ────────────────────────
+for tpl in BUILD_DIR.glob("*.yaml"):
+    cls = tpl.stem
+    print(f"\n🔨 Generating JSON Schema for {cls}")
+    schema_str = run_capture([
+        "gen-json-schema",
+        "--top-class", cls,
+        "--inline", "--no-metadata",
+        "--title-from=title",
+        "--closed", str(tpl)
+    ])
+    if not schema_str:
+        print(f"⚠️  Skipping {cls}")
+        continue
+
+    try:
+        raw = json.loads(schema_str)
+    except json.JSONDecodeError as e:
+        print(f"❌ Invalid JSON for {cls}: {e}")
+        continue
+
+    # set ID + title
+    raw["$id"]   = (
```

Comment:
This would need to be `https://repo-prod.prod.sagebase.org/repo/v1/schema/type/registered/org.synapse.nf-{classid}` since it would be under nf org